keps/sig-network/4770-endpointslice-controller-flexibility/README.md

```diff
@@ -0,0 +1,481 @@
+
+### Goals
+
+* Provide the ability to disable the Kubernetes EndpointSlice controller for particular services.
+* Extend and enhance the EndpointSlice Reconciler module to make it generic.
+
+### Non-Goals
+
+* Change / Replace / Deprecate the existing behavior of the Kubernetes EndpointSlice controller.
+* Introduce additional supported types of the EndpointSlice controllers/Reconciler as part of Kubernetes.
+* Modify the Service / EndpointSlice Specs.
+
+## Proposal
+
+#### Well-Known Label
+
+`service.kubernetes.io/endpoint-controller-name` will be added as a well-known label applying on the Service object. When set on a service, the Endpoint, EndpointSlice, and EndpointSlice Mirroring for that service will be disabled, thus EndpointSlices for this service will not be created by the Kubernetes Controller Manager. 
+
+#### Flexibility on the EndpointSlice Reconciler Module
+
+The reconciler structure will support more features to cover the requirements set by the current behavior of the EndpointSlice Mirroring Controller (e.g.: placeholder concept does not exist in the Mirroring Controller).
+
+The `Reconcile` function definition will be changed to accept general types of data (not specific to Services) and the list of features the endpointslices being reconciled will support (e.g. Traffic Distribution). Functions for specific types (Service/Pods for the EndpointSlice Controller and Endpoints for the EndpointSlice Mirroring Controller) generating data to pass to the `Reconcile` function will be provided by the EndpointSlice Reconciler module.
+
+The Metrics and its cache between the EndpointSlice Reconciler and EndpointSlice Mirroring Reconciler are similar but not identical. They will be merged together to provide a re-usable and non-type specific metrics package.
+
+The EndpointSlice controller will be adapted accordingly to support the new way of using the EndpointSlice Reconciler.
+
+Finally, the EndpointSlice Mirroring reconciler will be discarded from `pkg/controller/endpointslicemirroring` and the EndpointSlice Mirroring Controller will use the EndpointSlice reconciler from staging with the appropriate configuration.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a Cloud Native Network Function (CNF) vendor, some of my services are handled by custom Service-Proxies/Gateways over secondary networks provided by, for example, [Multus](https://github.com/k8snetworkplumbingwg/multus-cni). IPs configured in the service and registered by the EndpointSlice controller must be only the secondary IPs provided by the secondary network provider (e.g. [Multus](https://github.com/k8snetworkplumbingwg/multus-cni), also see [KEP-3698](https://github.com/kubernetes/enhancements/issues/3698)).
```

Comment:
```suggestion
As a Cloud Native Network Function (CNF) vendor, some of my services are handled by custom Service-Proxies/Gateways over secondary networks provided by, for example, [Multus](https://github.com/k8snetworkplumbingwg/multus-cni). IPs configured in the service and registered by the EndpointSlice controller must be only the secondary IPs provided by the secondary network provider (e.g. [Multus](https://github.com/k8snetworkplumbingwg/multus-cni).
```

KEP-3698 is not approved and has still a lot of pending comments without answer, so using it as foundation of this KEP is not ideal

Reply:
I removed it.